core/src/main/scala/chisel3/Module.scala

```diff
@@ -574,6 +574,21 @@ package experimental {
       _ports.toSeq
     }
 
+    /** Get IOs that are currently bound to this module.
+      */
+    private[chisel3] def getIOs: Seq[Data] = {
+      _ids.flatMap { id =>
+        id match {
+          case (data: Data) if data.isSynthesizable =>
+            data.topBinding match {
+              case PortBinding(_) => Some(data)
+              case _              => None
+            }
+          case _ => None
+        }
+      }.toSeq
```

Comment:
```suggestion
      _ports.toSeq
```
You can just use `this._ports`!